rest/response.go

```diff
@@ -100,9 +102,11 @@ func (f DefaultResponseFormatter) FormatList(ctx context.Context, headers http.H
 			}
 			if item.ETag != "" {
 				d["_etag"] = item.ETag
+				hash.Write([]byte(item.ETag))
 			}
 			payload[i] = d
 		}
+		headers.Set("ETag", `W/"`+fmt.Sprintf("%x", hash.Sum(nil))+`"`)
```

Comment:
I would suggest doing a strong E-tag in the `ResponseSender` instead:

https://github.com/rs/rest-layer/blob/2c4c847a7e4d13e1245c93990665aaeb1a35cfe2/rest/response.go#L57

Then we can just do `hash.Write(j)` in one line - it's not affected by some of the resource embedding would give us, as discussed in #157.

If we are worried about the performance, I suggest we benchmark it with a few different response sizes. If it's just a few milliseconds, or less, I don't think we should worry.

Reply:
Lets see what @rs will say. For me, hash of hashes in this case is even stronger ETag, because included `_etag` field will be different even for items fields changed outside the projection.

Reply:
Actually... it appears the list-view `E-tag` is only include if `skipBody == false`. Will the `E-tag` here even be included on `HEAD` requests?

Maybe not so easy to do this Hash in the sender neither btw, as it doesn't get passed in the `skipBody` parameter.

Reply:
Yes, FormatList is the right place.

Reply:
> Lets see what @rs will say. For me, hash of hashes in this case is even stronger ETag, because included _etag field will be different even for items fields changed outside the projection.

Well, there are _no_ includede _etag fields atm, except for the list-view root-resources, and then it doesn't matter. Anyway, as pointed out, not trivial to do it in the sender -- unless we want to change the interfaces.

Reply:
For now, maybe just leave it as a weak E-tag of tags, but move it out of the `!if skipBody`:

```go
hash := md5.New()
for i, item := range l.Items {
        if item.Etag != "" {
                hash.Write([]byte(item.ETag))
        }
}

```

@rs, what do you think about that? We can look at handling embedded sub-resources and potentially strong E-tags later if we want to.